rust/src/interpreter/gas.rs

```diff
@@ -26,34 +29,31 @@ pub(super) fn consume_positive_value_cost(
 pub(super) fn consume_value_to_empty_account_cost(
     value: &u256,
     addr: &Address,
-    context: &mut ExecutionContext,
-    gas_left: &mut u64,
+    state: &mut RunState,
 ) -> Result<(), StatusCode> {
-    if *value != u256::ZERO && !context.account_exists(addr) {
-        consume_gas(gas_left, 25000)?;
+    if *value != u256::ZERO && !state.context.account_exists(addr) {
+        consume_gas(&mut state.gas_left, 25000)?;
     }
     Ok(())
 }
 
 #[inline(always)]
 pub(super) fn consume_address_access_cost(
-    gas_left: &mut u64,
     addr: &Address,
-    context: &mut ExecutionContext,
-    revision: Revision,
+    state: &mut RunState,
 ) -> Result<(), StatusCode> {
-    let tx_context = context.get_tx_context();
-    if revision < Revision::EVMC_BERLIN {
+    let tx_context = state.context.get_tx_context();
+    if state.revision < Revision::EVMC_BERLIN {
         return Ok(());
     }
     if *addr != tx_context.tx_origin
             //&& addr != tx_context.tx_to // TODO
-            && !(revision >= Revision::EVMC_SHANGHAI && *addr == tx_context.block_coinbase)
-            && context.access_account(addr) == AccessStatus::EVMC_ACCESS_COLD
+            && !(state.revision >= Revision::EVMC_SHANGHAI && *addr == tx_context.block_coinbase)
```

Comment:
As in the other place where you compute access costs, it should not be necessary for you to explicitly check for the `block_coinbase` and other locations. The `context`'s `access_account` function should already know about those.